developers/blobstream-rollups.md

```diff
@@ -0,0 +1,399 @@
+---
+description: Learn how to build rollups that use Blobstream
+---
+
+## Introduction
+
+[Blobstream](https://blog.celestia.org/introducing-blobstream/)
+is the first data availability solution for EVM chains that securely
+scales with the number of users.
+It allows rollups to post their data on Celestia while proving their availability in the rollup settlement contract.
+
+This document will outline a few ways to build optimistic or zk-rollups that post their data to Celestia
+and use Blobstream to prove that data's availability.
+
+## Concepts
+
+### Share commitment
+
+The [share commitment](https://celestiaorg.github.io/celestia-app/specs/data_square_layout.html#blob-share-commitment-rules) is a commitment over the data
+contained in the [PayForBlob](https://github.com/celestiaorg/celestia-app/blob/v1.0.0-rc2/proto/celestia/blob/v1/tx.proto#L16-L31) transaction.
+This commitment allows
+proving that the corresponding data exists on Celestia [efficiently](https://github.com/celestiaorg/celestia-app/blob/main/docs/architecture/adr-011-optimistic-blob-size-independent-inclusion-proofs-and-pfb-fraud-proofs.md).
+
+#### Proof details
```

Comment:
do we really have 6 different proof types? if so, do users need to know about all 6? in this case, perhaps we should only focus on the proofs that the users need (the proofs returned by the api), then describe what each consists of if needed

Reply:
The reason I didn't include API is that we still don't have it... So I decided to go low level to explain the proofs. One other problem is that we don't have any document focusing on explaining the square layout in a beginner friendly way...

Reply:
thanks for clarifying! that makes sense and its okay to wait, but I do think we should try to simply things as much as possible given the proofs are so layered

Reply:
Once https://github.com/celestiaorg/celestia-app/pull/3245 is merged, we can link to it in these proofs to reduce confusion

Reply:
[0545d7f](https://github.com/celestiaorg/docs/pull/1489/commits/0545d7fba3f5ebd49c963507a25da8f9f0263acc) adds information on these proofs